src/main/java/com/amazon/opendistroforelasticsearch/sql/metrics/RollingCounter.java

```diff
@@ -0,0 +1,102 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.metrics;
+
+import com.amazon.opendistroforelasticsearch.sql.esdomain.LocalClusterState;
+import com.amazon.opendistroforelasticsearch.sql.plugin.SqlSettings;
+
+import java.time.Clock;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.concurrent.atomic.LongAdder;
+
+public class RollingCounter implements Counter<Long> {
+
+    private final long capacity;
+    private final long window;
+    private final long interval;
+    private final Clock clock;
+    private final ConcurrentSkipListMap<Long, Long> time2CountWin;
+    private final LongAdder count;
+
+    public RollingCounter() {
```

Comment:
please avoid duplicating code:
```java
public RollingCounter() {

  final long window = LocalClusterState.state().getSettingValue(SqlSettings.METRICS_ROLLING_WINDOW);
  final long interval = LocalClusterState.state().getSettingValue(SqlSettings.METRICS_ROLLING_INTERVAL);

  this(window, interval);
}
```

Reply:
I think this won't work, because the constructor call has to be the first, which means you can inline the window and interval calculations.

Reply:
Good suggestion!